src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -338,35 +338,24 @@ public override async Task ValidateIntegrityAsync(SignatureContent signatureCont
 #endif
         }
 
-        public override string GetContentHashForSignedPackage(CancellationToken token)
+        public override string GetContentHash(CancellationToken token, Func<string> fallbackHashGenerator = null)
```

Comment:
`fallbackHashGenerator` could you call it `GetUnsignedPackageHash`?

or even why even need a func? at least in current impl, you're just passing throw pre existing hash why not just pass a `string unsignedPackageHash`?

Reply:
There are two scenarios where the `fallbackHashGenerator` is used. In one of them we have the `unsignedPackageHash` already calculated (and a file path to the a temp hash file) and in the other we have only the file path to the sha512 hash file. The idea of this function is to avoid the disk hit when its not needed. Therefore in one place we can just return the hash without reading the file, but in the other scenario we need to read the file.

Reply:
then how about changing the func name to `GetUnsignedPackageHash`?